os/fs/procfs/fs_procfsproc.c

```diff
@@ -381,24 +382,24 @@ static ssize_t proc_entry_stat(FAR struct proc_file_s *procfile, FAR struct tcb_
 	peak_stack = -1;
 #endif
 
-	curr_heap = -1;
-	peak_heap = -1;
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
-	if (tcb->pid == 0) {
-		/* Idle task normally uses heap with index 0. */
-		heap = mm_get_heap_with_index(0);
-	} else {
-		heap = mm_get_heap(tcb->stack_alloc_ptr);
-	}
-	if (heap == NULL) {
-		return -1;
-	}
-
+	curr_heap = 0;
```

Comment:
this 0 initialization could refine the logic of += using in for loop, but it's changing the default -1 value when there is no alloc_size for current tcb. how about initialize it to -1 as before, and check it after for loop to revise the -1

Reply:
If CONFIG_DEBUG_MM_HEAPINFO is not defined, the default value is -1 (feature not support). But if it's defined, -1 is an illegal value (the minimal value is 0).